Source/WebCore/html/parser/HTMLDocumentParserFastPath.cpp

```diff
@@ -0,0 +1,1053 @@
+                return didFail(HTMLFastPathResult::FailedParsingCharacterReference);
+
+            for (unsigned i = 0; i < entity.length(); ++i)
+                out.append(entity[i]);
+        }
+    }
+
+    void didFail(HTMLFastPathResult result)
+    {
+        // This function may be called multiple times. Only record the result the first time it's called.
+        if (m_parsingFailed)
+            return;
+
+        m_parseResult = result;
+        m_parsingFailed = true;
+    }
+
+    template<class ResultType> ResultType didFail(HTMLFastPathResult result, ResultType res)
+    {
+        didFail(result);
+        return res;
+    }
+
+    Char peekNext()
+    {
+        ASSERT(m_position <= m_end);
+        if (m_position == m_end) {
+            didFail(HTMLFastPathResult::FailedEndOfInputReached);
+            return '\0';
+        }
+        return *m_position;
+    }
+
+    Char consumeNext()
+    {
+        if (m_position == m_end)
+            return didFail(HTMLFastPathResult::FailedEndOfInputReached, '\0');
+        return *(m_position++);
+    }
+
+    template<class ParentTag> void parseChildren(ContainerNode& parent)
+    {
+        while (true) {
+            std::pair<CharSpan, UCharSpan> text = scanText();
+            if (m_parsingFailed)
+                return;
+
+            ASSERT(text.first.empty() || text.second.empty());
+            if (!text.first.empty()) {
+                if (text.first.size() >= Text::defaultLengthLimit)
+                    return didFail(HTMLFastPathResult::FailedBigText);
+                parent.parserAppendChild(Text::create(m_document, String(text.first.data(), static_cast<unsigned>(text.first.size()))));
+            } else if (!text.second.empty()) {
+                if (text.second.size() >= Text::defaultLengthLimit)
+                    return didFail(HTMLFastPathResult::FailedBigText);
+                parent.parserAppendChild(Text::create(m_document, String(text.second.data(), static_cast<unsigned>(text.second.size()))));
+            }
+            if (m_position == m_end)
+                return;
+            ASSERT(*m_position == '<');
+            ++m_position;
+            if (peekNext() == '/') {
+                // We assume that we found the closing tag. The tagName will be checked by the caller `parseContainerElement()`.
+                return;
+            }
+            if (++m_elementDepth == Settings::defaultMaximumHTMLParserDOMTreeDepth)
+                return didFail(HTMLFastPathResult::FailedMaxDepth);
+            auto child = ParentTag::parseChild(*this);
+            --m_elementDepth;
+            if (m_parsingFailed)
+                return;
+            ASSERT(child);
+            parent.parserAppendChild(*child);
+        }
+    }
+
+    Attribute processAttribute(CharSpan nameSpan, std::pair<CharSpan, UCharSpan> valueSpan)
+    {
+        QualifiedName name = HTMLNameCache::makeAttributeQualifiedName(nameSpan);
```

Comment:
Why can’t this just call `makeQualifiedName`? Is there some kind of abstraction involved?

Reply:
I couldn't find `makeQualifiedName()`? Unless you mean the private function on `HTMLNameCache`? It is private so I can't currently call it. The HTMLNameCache was introduced fairly recently by Cameron I think as a Speedometer optimization. I matched what the regular parser is doing.